lua/hbac/command/subcommands.lua

```diff
@@ -0,0 +1,45 @@
+local autocommands = require("hbac.autocommands")
+local hbac_config = require("hbac.setup").opts
+local state = require("hbac.state")
+local utils = require("hbac.utils")
+
+local M = {}
+
+M.close_unpinned = function()
+	local curbufnr = vim.api.nvim_get_current_buf()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		local windows_with_buf = vim.fn.win_findbuf(bufnr)
+		if bufnr ~= curbufnr and not state.is_pinned(bufnr) and #windows_with_buf == 0 then
+			hbac_config.close_command(bufnr)
+		end
+	end
+end
+
+M.toggle_pin = function()
+	local bufnr = vim.api.nvim_get_current_buf()
+	local pinned_state = state.toggle_pin(bufnr) and "pinned" or "unpinned"
+	return bufnr, pinned_state
+end
+
+M.pin_all = function()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		state.pinned_buffers[bufnr] = state.pin_all
+	end
+	state.pin_all = not state.pin_all
+	return state.pin_all
+end
+
+M.toggle_autoclose = function()
+	if state.autoclose_enabled then
+		state.autoclose_enabled = false
+		autocommands.autoclose.disable()
+	else
+		state.autoclose_enabled = true
+		autocommands.autoclose.setup()
+	end
+	return state.autoclose_enabled
```

Comment:
I like early returns. What do you think? Is this more / less readable?
```suggestion
    state.autoclose_enabled = not state autoclose_enabled
	if state.autoclose_enabled then
		autocommands.autoclose.setup()
		return state.autoclose_enabled
	end
	autocommands.autoclose.disable()
	return state.autoclose_enabled
```